### PR TITLE
fix type handling for clocktime and step_size

### PR DESCRIPTION
**Pull request by stevebachmeier**
### Description

- *Category*:  other
- *JIRA issue*: https://jira.ihme.washington.edu/browse/MIC-5480

### CODE REVIEW

**Review by hussain-jafari — APPROVED** [review]
approved assuming commented out code will be removed

File: src/vivarium/framework/event.py
Comment (rmudambi) [inline src/vivarium/framework/event.py:136]: Remove these commented lines

File: CHANGELOG.rst
Comment (hussain-jafari) [inline CHANGELOG.rst:1]: is this intended to be tomorrow?

**Review by rmudambi — APPROVED** [review]

File: CHANGELOG.rst
Comment (stevebachmeier) [reply CHANGELOG.rst:1]: Yes just b/c I rarely get approvals this late in the day

**Review by albrja — APPROVED** [review]

File: src/vivarium/framework/event.py
Comment (rmudambi) [inline src/vivarium/framework/event.py:124]: Apparently you can just do datetime and timedelta, since the pandas types inherit from them

**Review by albrja — APPROVED** [review]

**Pull request merged**
